x-pack/plugins/dataset_quality/public/controller/types.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Observable } from 'rxjs';
+import { Direction } from '../hooks';
+import { DatasetQualityControllerStateService } from '../state_machines/dataset_quality_controller';
+
+export interface DatasetQualityController {
+  state$: Observable<DatasetQualityPublicState>;
+  service: DatasetQualityControllerStateService;
+}
+
+export interface DatasetQualityTableOptions {
+  page?: number;
+  rowsPerPage?: number;
+  sort?: {
+    field: string;
+    direction: Direction;
+  };
+}
+
+export interface DatasetQualityFlyoutOptions {
+  dataset?: {
+    rawName: string;
+    type: string;
+    name: string;
+    namespace: string;
+    title: string;
+    integration?: {
+      name: string;
+      title: string;
+      version: string;
+    };
+  };
+}
```

Comment:
**question**: this looks familiar, is there any type derived from rt that we can use to extract this type?

Reply:
Cannot be derived from anywhere else 😅

Reply:
Aren't these the same properties of a [DataStreamStats](https://github.com/elastic/kibana/blob/main/x-pack/plugins/dataset_quality/common/data_streams_stats/data_stream_stat.ts) instance? If so, you can use the class as a type and Pick the props you need.

Reply:
The problem with that class comes with the `Integration` type, where title could be nullable, while here it shouldn't

Reply:
Addressed in [2d6248c](https://github.com/elastic/kibana/pull/174906/commits/2d6248c449f44dc2abfaee6db34d484f0c742a9c).